semgrep-core/src/tainting/Dataflow_tainting.ml

```diff
@@ -160,23 +171,71 @@ let merge_source_sink_mvars env source_mvars sink_mvars =
     (* The union of both sets, but taking the sink mvars in case of collision. *)
     sink_biased_union_mvars source_mvars sink_mvars
 
+let labels_in_taint taints : LabelSet.t =
+  taints |> Taints.to_seq
+  |> Seq.filter_map (fun (t : T.taint) ->
+         match t.orig with
+         | Src src ->
+             let _, ts = T.pm_of_trace src in
+             Some ts.Rule.label
+         | Arg _ -> None)
+  |> LabelSet.of_seq
+
+let rec eval_label_requires ~labels e =
```

Comment:
maybe add a (* coupling: if you modify the code here, you probably need to modify Parse_rule.parse_require_expression *)